src/lualib/Decorate.ts

```diff
@@ -11,8 +11,17 @@ function __TS__Decorate(this: void, decorators: Function[], target: {}, key?: st
 
             if (key === undefined) {
                 result = decorator(result);
-            } else if (desc !== undefined) {
-                result = decorator(target, key, result);
+            } else if (desc === true) {
+                const value = rawget(target, key);
+                let descriptor: PropertyDescriptor;
+                if (value === undefined) {
+                    descriptor = __TS__ObjectGetOwnPropertyDescriptor(target, key);
+                } else {
+                    descriptor = __TS__CloneDescriptor({ configurable: true, writable: true, value });
+                }
+                __TS__SetDescriptor(target, key, { ...descriptor, ...decorator(target, key, descriptor) });
```

Comment:
`__TS__SetDescriptor` has a quite large performance impact. It would be nice to avoid using actual descriptors, unless it's necessary

```ts
const isSimpleValue = descriptor.configurable === true && descriptor.writable === true && !descriptor.get && !descriptor.set;
if (isSimpleValue) {
	target[key] = descriptor.value;
} else {
	__TS__SetDescriptor(target, key, descriptor);
}
```

Reply:
Descriptors shouldn't exist until the user decides to put them in via decorators/getters/setters or `Object.defineProperty`

An issue here is if the user explicitly sets the descriptor they may not be able to see it via `Object.getOwnPropertyDescriptor` since it is optimised away

So should we implement and note this as a bug/quirk of descriptors in TSTL?

Reply:
Or maybe an opt-in `--optimizeDescriptors` option that engages this behaviour for generated code so users can have a choice

Reply:
> An issue here is if the user explicitly sets the descriptor they may not be able to see it via Object.getOwnPropertyDescriptor since it is optimised away

That would be an issue anyway, because non-decorated members should have descriptors too. So I think it's fine to accept it as an another point of difference